src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,232 @@
+    super();
+    this.options_ = options;
+  }
+
+  transformClassElements_(tree) {
+    let superClass = this.transformAny(tree.superClass);
+
+    let elements = [];
+    let initInstanceVars = [], initStaticVars = [];
+    let constructor;
+
+    tree.elements.forEach((tree) => {
+      let initVars;
+      if (tree.isStatic) {
+        initVars = initStaticVars;
+      } else {
+        initVars = initInstanceVars;
+      }
+
+      switch (tree.type) {
+        case GET_ACCESSOR:
+          let getter = new GetAccessor(tree.location, tree.isSatic, tree.name,
+              tree.typeAnnotation, tree.annotations, tree.body);
+          elements.push(this.transformAny(getter));
+          break;
+
+        case SET_ACCESSOR:
+          let setter = new SetAccessor(tree.location, tree.isSatic, tree.name,
+              tree.parameterList, tree.annotations, tree.body);
+          elements.push(this.transformAny(setter));
+          break;
+
+        case PROPERTY_METHOD_ASSIGNMENT:
+          if (!tree.isStatic && propName(tree) === CONSTRUCTOR) {
+            constructor = tree;
+          } else {
+            let pma = new PropertyMethodAssignment(tree.location, tree.isStatic,
+                tree.functionKind, tree.name, tree.parameterList,
+                tree.typeAnnotation, tree.annotations, tree.body,
+                tree.debugName);
+            elements.push(this.transformAny(pma));
+          }
+          break;
+
+        case PROPERTY_VARIABLE_DECLARATION:
+          this.transformAny(tree);
+          if (tree.initializer !== null) {
+            initVars.push(tree);
+          }
+          break;
+
+        default:
+          throw new Error(`Unexpected class element: ${tree.type}`);
+      }
+    });
+
+    let initStatements = getInstanceInitStatements(initInstanceVars);
+
+    if (!constructor) {
+      constructor = this.getDefaultConstructor_(tree, initStatements);
+    } else {
+      if (this.options_.memberVariables) {
+        constructor = this.appendInstanceInitializers_(constructor,
+            initStatements, tree.superClass);
+      }
+    }
+
+    elements.push(constructor);
+
+    return {
+      elements,
+      initStaticVars,
+    };
+  }
+
+  /**
+   * Transforms a single class declaration
+   *
+   * @param {ClassDeclaration} tree
+   * @return {ParseTree}
+   */
+  transformClassDeclaration(tree) {
+    let {
+      elements,
+      initStaticVars,
+    } = this.transformClassElements_(tree);
+
+    let statements = createStaticInitializerStatements(tree.name, initStaticVars);
+
+    statements = prependStatements(statements,
+        createClassDeclaration(tree.name, tree.superClass, elements));
+
+    return new AnonBlock(null, statements);
+  }
+
+  getDefaultConstructor_(tree, initStatements) {
+    let constructorParams = createEmptyParameterList();
+    let constructorBody;
+    if (tree.superClass) {
+      let statement = parseStatement `super(...arguments)`;
```

Comment:
@arv this would trigger an error:
`null:1:1: super is only allowed in methods`

What would be the best way to handle this, create the super expression by hand ?